src/Git/PickLastMinorVersionFromCollection.php

```diff
@@ -19,8 +18,6 @@
      */
     public function forVersions(VersionsCollection $versions) : Version
     {
-        Assert::that($versions->count())->greaterOrEqualThan(1);
-
         $versions->sort(VersionsCollection::SORT_DESC);
```

Comment:
Relying on internal sorting is indeed risky, and I'm sure @nikolaposa will also make these collections immutable at some point, so it is probably a good idea to cast everything to an `array` via `iterator_to_array()` and operate on that afterwards.

Reply:
@Ocramius TBH I even planned to remove `VersionsCollection` class entirely in a newest 3.0.0 release, thinking it was unnecessary and nobody was using it. :scream: That's why I didn't give enough attention to its design. Feel free to [open an issue](https://github.com/nikolaposa/version/issues) describing fallacies of the class and I'll see what can be done. Of course, if you already have some idea for fixing it, open a PR along the way, I would be glad to accept it.